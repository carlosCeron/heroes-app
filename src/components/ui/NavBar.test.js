
import { mount } from 'enzyme';
import { Navbar } from './NavBar';
import { AuthContext } from './../../auth/AuthContext';
import { MemoryRouter, Router } from 'react-router-dom';
import { types } from '../../types/types';
describe('probando <NavBar /> component', () => {

  const historyMock = {
    push: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
    replace: jest.fn()
  }

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: 'Pedro'
    }
  }


  const wrapper = mount( 
    <AuthContext.Provider value={contextValue} >
      <MemoryRouter>
        <Router history={historyMock} >
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  )

  test('debe mostrar el navbar correctamente', () => {

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Pedro');

  });

  test('debe llamar el logout y usar history', () => {

    wrapper.find('button').prop('onClick')();

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.logout
    });

    expect( historyMock.replace ).toHaveBeenCalledWith('/login');

  });
});