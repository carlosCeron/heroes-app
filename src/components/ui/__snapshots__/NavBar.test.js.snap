// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <NavBar /> component debe mostrar el navbar correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 text-right"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <a
        className="nav-item nav-link text-info"
      >
        Pedro
      </a>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
